msupply-foundation/open-msupply
# 3299 link original shipment to return

Fixes #3299

# 👩🏻‍💻 What does this PR do? 
 
When a return is created from a shipment, store that original shipment ID so we can link to that shipment from the return's "Related Documents":
<img width="329" alt="Screenshot 2024-03-26 at 12 25 56 PM" src="https://github.com/msupply-foundation/open-msupply/assets/55115239/f9db2863-40c2-40f4-9be5-041c61d56c04">


# 🧪 How has/should this change been tested? 


You'll need latest develop of mSupply running!

- Find an inbound shipment that was automatically created (i.e. has a link to an outbound shipment in another store)
- Create a return from some of those inbound shipment lines
- In the Outbound Return detail view, see the inbound shipment in Related Documents in the detail tab
- Make sure Outbound Return is at least Picked, so it will be transferred to the other store
- Switch to that supplying store
- Go to Inbound Returns, see the new return
- In the Inbound Return detail view side panel, you should see a related Outbound Shipment - this should be the original outbound shipment!

Manual returns have no related documents.

## 💌 Any notes for the reviewer?


Note, we don't currently have any relationship from the shipment back to the return. I.e. in the Outbound Shipment, there is no "related returns". This was agreed to in a stakeholder meeting (as potentially we could have many returns from a shipment - different lines at different times...) but just flagging that here!
